SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Storage/Storage.cs

```diff
@@ -149,10 +185,27 @@ await _storageAdapter.DeleteAsync<LogEntry>(entry => entry.Channel == channelNam
         /// <exception cref="StorageException"/>
         public async Task<int> CountLogsAsync(string channelName)
         {
-            using (await _taskLockSource.GetTaskLockAsync().ConfigureAwait(false))
+            var task = new Task<int>(() =>
+            {
+                return _storageAdapter.CountAsync<LogEntry>(entry => entry.Channel == channelName)
+                    .Result;
+            });
+            try
+            {
+                _queue.Add(task);
+            }
+            catch (InvalidOperationException)
             {
-                return await _storageAdapter.CountAsync<LogEntry>(entry => entry.Channel == channelName)
-                    .ConfigureAwait(false);
+                throw new StorageException("The operation has been cancelled");
+            }
+            try
+            {
+                await FlushQueue().ConfigureAwait(false);
+                return await task.ConfigureAwait(false);
+            }
+            catch (AggregateException e)
```

Comment:
>catch (AggregateException e) [](start = 12, length = 28)

Two questions: (1) which line throws AggregateException? under what condition? (2) Why AggregateException there's no try-catch for other places where FlushQueue() is called in this file?